### PR TITLE
Fix associations

### PR DESCRIPTION
**Pull request by andrewsouthard1**
Fix user associations, created rspec test in course_spec

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -9,9 +9,7 @@ class User < ApplicationRecord
   belongs_to :local_school, optional: true
   belongs_to :course, optional: true
   has_many :charges
-  has_many :course_registrations
-  has_many :courses, through: :course_registrations
-
+  belongs_to :courses
```
Comment (trevordjones) [inline app/models/user.rb]: singular

File: spec/factories/courses.rb [hunk 2]
```diff
@@ -1,4 +1,12 @@
 FactoryGirl.define do
+  factory :user do 
+    first_name "John"
+     last_name "Smith"
+     email "js@email.com"
+     # course_id course.id
```
Comment (trevordjones) [inline spec/factories/courses.rb]: if we don't need it, don't include it. Do we need it?

File: app/models/user.rb (hunk 1 above)
Comment (trevordjones) [inline app/models/user.rb]: Don't other models need to be updated? Make sure all these associations are working

File: spec/factories/courses.rb (hunk 2 above)
Comment (jameschou93) [inline spec/factories/courses.rb]: We are using it in course_spec.rb
`  it "has many users" do
    course = FactoryGirl.build(:course)
    john = FactoryGirl.build(:user)
    tom = FactoryGirl.build(:user, first_name: "tom")
    course = FactoryGirl.create(:course)
    
    course.users << john << tom 
    expect(course.users[0].id).to eq(john.id)
  end`

Comment (jameschou93) [inline spec/factories/courses.rb]: Using it in course_spec.rb
`  it "has many users" do
    course = FactoryGirl.build(:course)
    john = FactoryGirl.build(:user)
    tom = FactoryGirl.build(:user, first_name: "tom")
    course = FactoryGirl.create(:course)
    course.users << john << tom 
    expect(course.users[0].id).to eq(john.id)
  end`

Comment (jameschou93) [inline spec/factories/courses.rb]: screen shot 2017-06-22 at 7 49 14 pm

Using it in course_spec.rb

File: app/models/user.rb (hunk 1 above)
Comment (jameschou93) [inline app/models/user.rb]: Made course singular. There aren't any models that use course_registration or course_registration association.

File: spec/rails_helper.rb [hunk 3]
```diff
@@ -29,6 +29,8 @@
 RSpec.configure do |config|
   # Remove this line if you're not using ActiveRecord or ActiveRecord fixtures
   config.fixture_path = "#{::Rails.root}/spec/fixtures"
+  
+  #config for
```
Comment (trevordjones) [inline spec/rails_helper.rb]: remove this

File: spec/models/course_spec.rb [hunk 4]
```diff
@@ -12,19 +12,12 @@
...
-      course_id: course.id
-    )
-    jeff = User.create(
-      first_name: "Jeff",
-      last_name: "Sessions",
-      email: "jjj@email.com",
-      course_id: course.id
-    )
+    
+    course.users << john << tom
```
Comment (trevordjones) [inline spec/models/course_spec.rb]: The way you test a `has_many` is like this:
`expect(Course.reflect_on_association(:users).macro).to eq(:has_many)` - so we can simplify this

File: spec/factories/courses.rb [hunk 5]
```diff
@@ -1,4 +1,11 @@
 FactoryGirl.define do
+  factory :user do 
+    first_name "John"
+     last_name "Smith"
+     email "js@email.com"
+     password "password"
+  end 
+
```
Comment (trevordjones) [inline spec/factories/courses.rb]: Create a different file for a factory user. This is a courses. Much like you wouldn't add the `User` class to the file that has the `Course` class. Also, try using the faker gem for these values

File: spec/rails_helper.rb (hunk 3 above)
Comment (andrewsouthard1) [inline spec/rails_helper.rb]: We have removed the comment on line 33

File: spec/models/course_spec.rb (hunk 4 above)
Comment (andrewsouthard1) [inline spec/models/course_spec.rb]: "has many users" test has now been refactored with the reflect_on_association method

**Pull request merged**